### PR TITLE
igl | opengl | The glUniformBlockBinding() function only needs to be called once for each program

### PR DESCRIPTION
**Pull request by vinsentli**
The glUniformBlockBinding() function only needs to be called once for each program and does not need to be called every frame.

### CODE REVIEW

File: src/igl/opengl/RenderPipelineState.h [hunk 1]
```diff
@@ -87,6 +87,7 @@ class RenderPipelineState final : public WithContext, public IRenderPipelineStat
   std::vector<int> activeAttributesLocations_;
   BlendMode blendMode_ = {GL_FUNC_ADD, GL_FUNC_ADD, GL_ONE, GL_ZERO, GL_ONE, GL_ZERO};
   bool blendEnabled_ = false;
+  bool hasLinkUniformBlockBindingPoint_ = false;
```
Comment (AmesingFlank) [inline src/igl/opengl/RenderPipelineState.h:90]: nit: let's call this `uniformBlockBindingPointSet_`

**Review by AmesingFlank — COMMENTED** [review]
There is an non-const
```
std::unordered_map<int, size_t>& uniformBlockBindingMap();
```
method exposed in `RenderPipelineState`, which makes this change not super-safe. However, it doesn't look like that method is used anywhere, so if you also remove that method in this PR, this is more ore less OK.

But also, just wanted to understand your motivation for making this change -- do you see any bugs/perf-issues caused by setting uniform block binding points multiple times?

File: src/igl/opengl/RenderPipelineState.h (hunk 1 above)
Comment (vinsentli) [reply src/igl/opengl/RenderPipelineState.h]: done

**Comment by vinsentli** [comment]
Calling multiple times does not cause any functional issues, but according to the relevant documentation, calling it once is sufficient.
https://learnopengl.com/code_viewer.php?code=advanced/advanced_glsl_uniform_buffer_objects

**Pull request closed without merging**